lib/matplotlib/widgets.py

```diff
@@ -2064,16 +2065,43 @@ def _on_key_release(self, event):
         """Key release event handler."""
 
     def set_visible(self, visible):
-        """Set the visibility of our artists."""
-        self.visible = visible
+        """Set the visibility of the selector artists."""
+        self._visible = visible
         for artist in self.artists:
             artist.set_visible(visible)
 
-    def clear(self):
-        """Clear the selection and set the selector ready to make a new one."""
+    def get_visible(self):
+        """Get the visibility of the selector artists."""
+        return self._visible
+
+    @property
+    def visible(self):
+        return self.get_visible()
+
+    @visible.setter
+    def visible(self, visible):
+        warnings.warn(
+            "The visible setter was deprecated in Matplotlib 3.6 and will "
+            "be removed two minor releases later. Use set_visible instead.",
+            _api.MatplotlibDeprecationWarning
+            )
+        self.set_visible(visible)
+
+    def clear(self, update=True):
```

Comment:
Does the user need to be able to set `update=False`? We don't have a `clear(update=...)` signature anywhere else in the library. If this is needed only internally, I suggest to instead do

```
def _clear_without_update(self):
    ...

def clear(self):
    self._clear_without_update()
    self.update()
```
and then use `_clear_without_update()` when you need it internally. That way, we can hide the update logic from the user.